src/Cache/Signature/FixerSignature.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Cache\Signature;
+
+use PhpCsFixer\Fixer\FixerInterface;
+
+final class FixerSignature
+{
+    private string $name;
+    private string $contentHash;
+
+    /**
+     * @var array<string, mixed>|bool
+     */
+    private $config;
+
+    private function __construct()
+    {
+    }
+
+    /**
+     * @param array<string, mixed>|bool $config
+     */
+    public static function fromInstance(FixerInterface $fixer, $config): self
+    {
+        $signature = new self();
+        $signature->name = $fixer->getName();
+        $signature->contentHash = md5(file_get_contents((new \ReflectionClass($fixer))->getFileName()));
+        $signature->config = \is_array($config) ? self::makeJsonEncodable($config) : $config;
+
+        return $signature;
+    }
+
+    /**
+     * @param array<string, mixed>|bool $config
+     */
+    public static function fromRawValues(string $name, string $contentHash, $config): self
+    {
+        $signature = new self();
+        $signature->name = $name;
+        $signature->contentHash = $contentHash;
+        $signature->config = \is_array($config) ? self::makeJsonEncodable($config) : $config;
+
+        return $signature;
+    }
+
+    public function equals(self $signature): bool
+    {
+        return $this->contentHash === $signature->getContentHash();
```

Comment:
Probably this should also check the config, so signatures are equal when both code and config is the same?

Reply:
Or I can just remove it since it's not used anywhere 😅. I've added `equals()` for the future usage, to extend it even more but at this point it's not needed.